lib/routes/discord/search.ts

```diff
@@ -0,0 +1,107 @@
+import path from 'node:path';
+
+import { config } from '@/config';
+import InvalidParameterError from '@/errors/types/invalid-parameter';
+import { Route } from '@/types';
+import { parseDate } from '@/utils/parse-date';
+import { art } from '@/utils/render';
+import ConfigNotFoundError from '@/errors/types/config-not-found';
+import { queryToBoolean } from '@/utils/readable-social';
+
+import { baseUrl, getGuild, searchGuildMessages, SearchGuildMessagesParams, HasType, VALID_HAS_TYPES } from './discord-api';
+
+export const route: Route = {
+    path: '/search/:guildId/:routeParams',
+    categories: ['social-media'],
+    example: '/discord/search/302094807046684672/content=friendly&has=image,video',
+    parameters: {
+        guildId: 'Guild ID',
+        routeParams: 'Search parameters, support content, author_id, mentions, has, min_id, max_id, channel_id, pinned',
+    },
+    features: {
+        requireConfig: [
+            {
+                name: 'DISCORD_AUTHORIZATION',
+                description: 'Discord authorization header',
+            },
+        ],
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: 'Guild Search',
+    maintainers: ['NekoAria'],
+    handler,
+};
+
+const parseSearchParams = (routeParams?: string): SearchGuildMessagesParams => {
+    if (!routeParams) {
+        return {};
+    }
+
```

Comment:
```suggestion
```
Unnecessary condition. Since the path is defined as `/search/:guildId/:routeParams`, requests without `routeParams` will be rejected by Hono.